week2/substitution/substitution.c

```diff
@@ -0,0 +1,74 @@
+#include <cs50.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+
+int main(int argc, string argv[])
+{
+  if (argc != 2)
+  {
+    printf("Usage: ./substitution key\n");
+
+    return 1;
+  }
+
+  string key = argv[1];
+  for (int i = 0; i < strlen(key); i++)
+  {
+    if (!isalpha(key[i]))
+    {
+      printf("only use letters\n");
+
+      return 1;
+    }
+  }
+
+  if (strlen(key) != 26)
+  {
+    printf("Key must contain 26 characters.\n");
+
+    return 1;
+  }
+
+  for (int i = 0; i < strlen(key); i++)
+  {
+    for (int j = i + 1; j < strlen(key); j++)
+    {
+      if (toupper(key[i]) == toupper(key[j]))
+      {
+        printf("use each letter only once\n");
+        
+        return 1;
+      }
+    }
+  }
+
+  string plaintext = get_string("plaintext: ");
+
+  for (int i = 0; i < strlen(key); i++)
+  {
+    if (islower(key[i]))
+    {
+      key[i] = key[i] - 32;
```

Comment:
```suggestion
      key[i] = toupper(key[i]);
```
This would be a more readable variant